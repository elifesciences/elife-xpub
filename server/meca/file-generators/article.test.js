const { createTables } = require('@pubsweet/db-manager')
const path = require('path')
const { db } = require('pubsweet-server')
const generateXml = require('./article')
const Replay = require('replay')
const sampleManuscript = require('./article.test.data')

// Ref: https://stackoverflow.com/questions/10265798/determine-project-root-from-a-running-node-js-application
const appRoot = process.cwd()
const { obfuscateEmail } = require(path.join(appRoot, '/scripts/obfuscate'))

const existingNames = [{ id: 1, first: 'J. Edward', last: 'Reviewer' }]

Replay.fixtures = `${__dirname}/../test/http-mocks`

describe('Article XML generator', () => {
  beforeAll(async () => {
    await createTables(true)
    await db.table('ejp_name').insert(existingNames)
  })

  it("doesn't choke on missing teams", async () => {
    const manuscriptWithoutTeams = { ...sampleManuscript, teams: [] }
    await generateXml(manuscriptWithoutTeams)
  })

  it('generates expected XML', async () => {
    let xml = await generateXml(sampleManuscript)
    xml = obfuscateEmail(xml)
    expect(xml).toMatchSnapshot()
  })
})
