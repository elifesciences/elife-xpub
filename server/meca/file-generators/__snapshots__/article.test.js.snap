// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article XML generator generates expected XML 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<!DOCTYPE article SYSTEM \\"JATS-journalpublishing1.dtd\\">
<article>
  <front>
    <journal-meta>
      <journal-id journal-id-type=\\"pmc\\"/>
      <journal-id journal-id-type=\\"publisher\\"/>
      <issn/>
      <publisher>
        <publisher-name/>
      </publisher>
    </journal-meta>
    <article-meta>
      <article-id pub-id-type=\\"manuscript\\">604e06ca-882d-4b5b-a147-e016893e60e9</article-id>
      <article-categories>
        <subj-group subj-group-type=\\"article_type\\">
          <subject>5</subject>
        </subj-group>
        <subj-group>
          <subj-group subj-group-type=\\"subject_areas\\">
            <subject>Biochemistry and Chemical Biology</subject>
          </subj-group>
        </subj-group>
        <subj-group>
          <subj-group subj-group-type=\\"subject_areas\\">
            <subject>Chromosomes and Gene Expression</subject>
          </subj-group>
        </subj-group>
      </article-categories>
      <title-group>
        <article-title>Test Manuscript</article-title>
      </title-group>
      <contrib-group content-type=\\"authors\\">
        <contrib contrib-type=\\"author\\" corresp=\\"yes\\">
          <name>
            <surname>User</surname>
            <given-names>Test</given-names>
          </name>
          <email>a94e85089c7efea17a0563010705cb29</email>
          <xref ref-type=\\"aff\\" rid=\\"aff6\\"/>
        </contrib>
      </contrib-group>
      <contrib-group content-type=\\"potential_senior_editors\\">
        <contrib contrib-type=\\"suggested_senior_editor\\">
          <name>
            <surname>Zoghbi</surname>
            <given-names>Huda</given-names>
          </name>
          <email>8ac3df4aa667c9217fe4d8bedfc06482</email>
          <xref ref-type=\\"aff\\" rid=\\"aff1\\"/>
        </contrib>
        <contrib contrib-type=\\"suggested_senior_editor\\">
          <name>
            <surname>Li</surname>
            <given-names>Wenhui</given-names>
          </name>
          <email>5ddd85b12237e714f490a9c6cf585566</email>
          <xref ref-type=\\"aff\\" rid=\\"aff2\\"/>
        </contrib>
        <contrib contrib-type=\\"opposed_senior_editor\\">
          <name>
            <surname>Struhl</surname>
            <given-names>Kevin</given-names>
          </name>
          <email>4f5cd09b1742dad17ffe48d0c33def8f</email>
          <xref ref-type=\\"aff\\" rid=\\"aff3\\"/>
        </contrib>
      </contrib-group>
      <contrib-group content-type=\\"potential_reviewing_editors\\">
        <contrib contrib-type=\\"suggested_reviewing_editor\\">
          <name>
            <surname>Zilberman</surname>
            <given-names>Daniel</given-names>
          </name>
          <email>08c6488ad00599649606fbb926a63f12</email>
          <xref ref-type=\\"aff\\" rid=\\"aff4\\"/>
        </contrib>
        <contrib contrib-type=\\"suggested_reviewing_editor\\">
          <name>
            <surname>Zhang</surname>
            <given-names>Hong</given-names>
          </name>
          <email>e69bf9f08f18063b3098e1cd63c26a23</email>
          <xref ref-type=\\"aff\\" rid=\\"aff5\\"/>
        </contrib>
        <contrib contrib-type=\\"opposed_reviewing_editor\\">
          <name>
            <surname>Yu</surname>
            <given-names>Hao</given-names>
          </name>
          <email>d3fa5c7081fecbc7cfc02be293834b86</email>
          <xref ref-type=\\"aff\\" rid=\\"aff0\\"/>
        </contrib>
      </contrib-group>
      <contrib-group content-type=\\"potential_reviewers\\">
        <contrib contrib-type=\\"suggested_reviewer\\">
          <name>
            <surname>Reviewer</surname>
            <given-names>J. Edward</given-names>
          </name>
          <email>db27cf2960d24994beeafa06b08a061d</email>
        </contrib>
        <contrib contrib-type=\\"suggested_reviewer\\">
          <name>
            <surname>de Reviewer</surname>
            <given-names>Frances</given-names>
          </name>
          <email>53d85c678da1478ed6815e1494bd15af</email>
        </contrib>
        <contrib contrib-type=\\"suggested_reviewer\\">
          <name>
            <surname>Reviewer</surname>
            <given-names>Gertrude</given-names>
          </name>
          <email>528a83f12f51aa69557835eb6c8eb3d2</email>
        </contrib>
        <contrib contrib-type=\\"opposed_reviewer\\">
          <name>
            <surname>Hoo</surname>
            <given-names/>
          </name>
          <email>d628cc24eba81f0a70dc05be9fffe22f</email>
        </contrib>
        <contrib contrib-type=\\"opposed_reviewer\\">
          <name>
            <surname>Who</surname>
            <given-names/>
          </name>
          <email>626ce554592c639a9d08b1e379eaf96d</email>
        </contrib>
      </contrib-group>
      <aff id=\\"aff0\\">
        <institution>National University of Singapore &amp; Temasek Life Sciences Laboratory</institution>
      </aff>
      <aff id=\\"aff1\\">
        <institution>Texas Children's Hospital</institution>
      </aff>
      <aff id=\\"aff2\\">
        <institution>National Institute of Biological Sciences</institution>
      </aff>
      <aff id=\\"aff3\\">
        <institution>Harvard Medical School</institution>
      </aff>
      <aff id=\\"aff4\\">
        <institution>John Innes Centre</institution>
      </aff>
      <aff id=\\"aff5\\">
        <institution>Institute of Biophysics Chinese Academy of Sciences</institution>
      </aff>
      <aff id=\\"aff6\\">
        <institution>University of eLife</institution>
      </aff>
      <author-notes>
        <fn>
          <p/>
        </fn>
      </author-notes>
      <pub-date>
        <day/>
        <month/>
        <year/>
      </pub-date>
      <related-article related-article-type=\\"companion\\">
        <article-title>asdassss</article-title>
      </related-article>
      <related-article related-article-type=\\"companion\\">
        <article-title>Another</article-title>
      </related-article>
      <custom-meta-group>
        <custom-meta>
          <meta-name>opposed-senior-editor-comment</meta-name>
          <meta-value>From another galaxy</meta-value>
        </custom-meta>
        <custom-meta>
          <meta-name>opposed-reviewing-editor-comment</meta-name>
          <meta-value>Not any more</meta-value>
        </custom-meta>
        <custom-meta>
          <meta-name>opposed-reviewer-comment</meta-name>
          <meta-value>Wandering days are over</meta-value>
        </custom-meta>
        <custom-meta>
          <meta-name>suggestions-conflict</meta-name>
          <meta-value>true</meta-value>
        </custom-meta>
        <custom-meta>
          <meta-name>previously-discussed</meta-name>
          <meta-value>Talked to bob about it</meta-value>
        </custom-meta>
        <custom-meta>
          <meta-name>previously-submitted</meta-name>
          <meta-value>Original Test Title</meta-value>
        </custom-meta>
        <custom-meta>
          <meta-name>created-by</meta-name>
          <meta-value>6d8cd1ce-15b6-46c1-b901-bc91598c8f2d</meta-value>
        </custom-meta>
      </custom-meta-group>
    </article-meta>
  </front>
</article>"
`;
